Python/Product/Profiling/Profiling/CommandBuilder.cs

```diff
@@ -0,0 +1,170 @@
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABILITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+
+
+namespace Microsoft.PythonTools.Profiling {
+    using System;
+    using System.Diagnostics;
+    using System.IO;
+    using System.Linq;
+    using System.Windows;
+    using Microsoft.PythonTools.Infrastructure;
+    using Microsoft.PythonTools.Interpreter;
+
+    internal class CommandBuilder {
+
+        /// <summary>
+        /// Constructs a <see cref="TargetCommand"/> based on the provided profiling target.
+        /// </summary>
+        public TargetCommand BuildCommandFromTarget(ProfilingTarget target) {
+            try {
+                if (target == null) {
+                    return null;
+                }
+
+                var pythonProfilingPackage = PythonProfilingPackage.Instance;
+                var joinableTaskFactory = pythonProfilingPackage.JoinableTaskFactory;
+
+                TargetCommand command = null;
+
+                joinableTaskFactory.Run(async () => {
+                    await joinableTaskFactory.SwitchToMainThreadAsync();
+
+                    var name = target.GetProfilingName(pythonProfilingPackage, out var save);
+                    var explorer = await pythonProfilingPackage.ShowPerformanceExplorerAsync();
+                    var session = explorer.Sessions.AddTarget(target, name, save);
+
+                    command = SelectCommandBuilder(target, session);
+
+                });
+
+                return command;
+            } catch (Exception ex) {
+                Debug.Fail($"Error building command: {ex.Message}");
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Generates a <see cref="TargetCommand"/> based on the profiling target and session details.
+        /// </summary>
+        private TargetCommand SelectCommandBuilder(ProfilingTarget target, SessionNode session) {
+            var projectTarget = target.ProjectTarget;
+            var standaloneTarget = target.StandaloneTarget;
+
+            if (projectTarget != null) {
+                return BuildProjectCommand(projectTarget, session);
+            } else if (standaloneTarget != null) {
+                return BuildStandaloneCommand(standaloneTarget, session);
+            }
+            return null;
+        }
+
+        private TargetCommand BuildProjectCommand(ProjectTarget projectTarget, SessionNode session) {
+            var solution = PythonProfilingPackage.Instance.Solution;
```

Comment:
Does this work if there is no solution? Like if the user is trying to profile files in folder view instead of project view?